include/oneapi/dpl/experimental/kt/single_pass_scan.h

```diff
@@ -0,0 +1,399 @@
+
+            // If we found a full value, we can stop looking at previous tiles. Otherwise,
+            // keep going through tiles until we either find a full tile or we've completely
+            // recomputed the prefix using partial values
+            if (__is_full_ballot_bits)
+                break;
+        }
+        return __running;
+    }
+
+    const uint32_t __tile_id;
+    _FlagType* __flags_begin;
+    _T* __full_vals_begin;
+    _T* __partial_vals_begin;
+    _AtomicFlagT __atomic_flag;
+    _AtomicValueT __atomic_partial_value;
+    _AtomicValueT __atomic_full_value;
+};
+
+template <typename _FlagType, typename _Type, typename _BinaryOp, typename _KernelName>
+struct __lookback_init_submitter;
+
+template <typename _FlagType, typename _Type, typename _BinaryOp, typename... _Name>
+struct __lookback_init_submitter<_FlagType, _Type, _BinaryOp,
+                                 oneapi::dpl::__par_backend_hetero::__internal::__optional_kernel_name<_Name...>>
+{
+    template <typename _StatusFlags, typename _PartialValues>
+    sycl::event
+    operator()(sycl::queue __q, _StatusFlags&& __status_flags, _PartialValues&& __partial_values, ::std::size_t __status_flags_size,
+               ::std::uint16_t __status_flag_padding) const
+    {
+        using _KernelName = __lookback_init_kernel<_Name..., _Type, _BinaryOp>;
+
+        return __q.submit([&](sycl::handler& __hdl) {
+            __hdl.parallel_for<_KernelName>(sycl::range<1>{__status_flags_size}, [=](const sycl::item<1>& __item) {
+                auto __id = __item.get_linear_id();
+                __status_flags[__id] =
+                    __id < __status_flag_padding ? _FlagType::__oob_status : _FlagType::__initialized_status;
+                __partial_values[__id] = oneapi::dpl::unseq_backend::__known_identity<_BinaryOp, _Type>;
+            });
+        });
+    }
+};
+
+template <::std::uint16_t __data_per_workitem, ::std::uint16_t __workgroup_size, typename _Type, typename _FlagType,
+          typename _KernelName>
+struct __lookback_submitter;
+
+template <::std::uint16_t __data_per_workitem, ::std::uint16_t __workgroup_size, typename _Type, typename _FlagType,
+          typename _InRng, typename _OutRng, typename _BinaryOp, typename _StatusFlags, typename _StatusValues,
+          typename _TileVals>
+struct __lookback_kernel_func
+{
+    using _FlagStorageType = typename _FlagType::_FlagType;
+    static constexpr std::uint32_t __elems_in_tile = __workgroup_size * __data_per_workitem;
+
+    _InRng __in_rng;
+    _OutRng __out_rng;
+    _BinaryOp __binary_op;
+    ::std::size_t __n;
+    _StatusFlags __status_flags;
+    ::std::size_t __status_flags_size;
+    _StatusValues __status_vals_full;
+    _StatusValues __status_vals_partial;
+    ::std::size_t __current_num_items;
+    _TileVals __tile_vals;
+
+    [[intel::reqd_sub_group_size(SUBGROUP_SIZE)]] void
```

Comment:
I think we can use `sycl::reqd_sub_group_size` here since it is a part of the SYCL 2020.

Reply:
Done.